File: oldc3/stack.cc
#include "stack.h"



int main(int argc, char* argv[]){
    stackNode<int> n(5);
    std::cout << n.getItem() << std::endl;
    stackNode<int> m(7);
    n.setNext(m);
    std::cout << n.getNext()->getItem() << std::endl;

    stack<int> s = stack<int>();
    s.push(2);
    std::cout << s.peek() << std::endl;
    s.push(6);
    std::cout << s.peek() << std::endl;
    s.pop();
    std::cout << s.peek() << std::endl;

    return 0;
}

File: chapter_three/src/chapter_three/Stack.java
package chapter_three;

public class Stack<T> {
	
	private StackNode<T> top;
	private int size;
	
	public Stack() {
		size = 0;
		top = null;
	}
	
	public void push(T item) {
		StackNode<T> node = new StackNode<T>(item);
		node.setNext(top);
		top = node;
		size++;
	}
	
	public T pop() {
		if (size == 0)
			return null;
		
		T item = top.getItem();
		top = top.getNext();
		size--;
		return item;
	}
	
	public Boolean isEmpty() {
		return size == 0;
	}
	

	public int getSize() {
		return size;
	}
	
	public T peek() {
		return top.getItem();
	}
	
	@Override
	public String toString() {
		
		if (size == 0)
			return null;
		StringBuilder sb = new StringBuilder();
		
		StackNode<T> temp = top;
		sb.append(temp.getItem().toString() + ", ");
		
		while (temp.hasNext()) {
			sb.append(temp.getNext().getItem().toString() + ", ");
			temp = temp.getNext();
		}
		
		return sb.toString();
		
	}
	
}
